src/activity_actor.h

```diff
@@ -176,6 +176,31 @@ class migration_cancel_activity_actor : public activity_actor
         static std::unique_ptr<activity_actor> deserialize( JsonIn &jsin );
 };
 
+class purify_water_activity_actor : public activity_actor
+{
+    private:
+        item &liquid;
```

Comment:
This should be an `item_location` not an `item &`, a simple reference like this can't be properly serialized and may easily be broken.